announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/entityprovider/AnnouncementEntityProviderImpl.java

```diff
@@ -277,9 +278,32 @@ private List<?> getAnnouncements(String siteId, Map<String,Object> params, boole
 	
 		for (Message m : announcements) {
 			AnnouncementMessage a = (AnnouncementMessage)m;
+			boolean showAnnouncement = false;
+			String selectedRoles = ((String) a.getProperties().get("selectedRoles"));
+			if (securityService.isSuperUser()) {
```

Comment:
if (securityService.isSuperUser() || selectedRoles == null) {
showAnnouncement = true;
} else {
}

Reply:
Let's be consistent with the action,

String ownerId = a.getAnnouncementHeader().getFrom().getId();
String userId = sessionManager.getCurrentSessionUserId();
boolean isOwner = ownerId.equals(userId);

if (isOwner || securityService.isSuperUser() || selectedRoles == null) {
} else {
}